client/snapctl.go

```diff
@@ -23,25 +23,17 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
-)
-
-// SnapCtlOptions holds the various options with which snapctl is invoked.
-type SnapCtlOptions struct {
-	// Context is a string used by snapd to determine the context of this call.
-	// (e.g. which handler should be used, etc.)
-	Context string `json:"context"`
 
-	// Args contains a list of parameters to use for this invocation.
-	Args []string `json:"args"`
-}
+	"github.com/snapcore/snapd/overlord/hookstate"
+)
 
 type snapctlOutput struct {
 	Stdout string `json:"stdout"`
 	Stderr string `json:"stderr"`
 }
 
 // RunSnapctl requests a snapctl run for the given options.
-func (client *Client) RunSnapctl(options SnapCtlOptions) (stdout, stderr []byte, err error) {
+func (client *Client) RunSnapctl(options hookstate.SnapCtlRequest) (stdout, stderr []byte, err error) {
```

Comment:
Moved this back to `client.SnapCtlOptions`, and used that throughout for consistency.